### DISCUSSION TITLE
Get cursor failed

### DISCUSSION THREAD

**Discussion by well0886** [appy-one/acebase-client#19] [Q&A] [answered]
When I run the example code for sychronizing remote changes to a local cache database, something is wrong with the cursor, the local cached database has to reload all the value from the server everytime. Here is a snippet of the code, could you please help me? 

 // Open local state database
const stateDb = AceBase.WithIndexedDB('localstate');
const cacheDb = AceBase.WithIndexedDB('offlinecache')

// Connect to remote server using a client with cache database
const client = new AceBaseClient({ 
    cache: { db: cacheDb },
    host: '192.168.1.111', 
    port:5757, 
    https: false, 
    dbname: 'mydbserver'
});

// Wait for both te be ready to use
await client.ready();
await stateDb.ready();

// Get last cursor we stored in our localstate database (if available)
let snap = await stateDb.ref('last_cursor').get();
const cursor = snap.val();

// Update the cache database with changes to the "products" collection since cursor.
// If cursor is null, the entire value will be loaded from the server and stored in cache.
const updateResult = await client.cache.update('products', cursor);

// Store the new cursor for next time
**_cursor = updateResult.new_cursor;     //here the cursor is null_**
await stateDb.ref('last_cursor').set(cursor);

// Monitor the "products" collection to keep cache up to date while running:
client.ref('products').on('mutations', snapshot => {
    // The snapshot contains an array with all effective mutations.
    // We don't have to do anything with these mutations here: the cache database will
    // automatically be updated with the changes, simply by monitoring this event
    // We CAN however use the new cursor we've been given now for the next startup sync!
    const context = snapshot.context();
    **_cursor = context.acebase_cursor;     //here the cursor is null too_**
    stateDb.ref('last_cursor').set(cursor);
    // Upon disconnect/reconnect cycles, the AceBaseClient will automatically
    // use the last cursor from this event to synchronize cache with the server.
    // If the data changes during reconnect sync, this event callback will run again 
    // and get the new cursor automatically
});

**Comment by appy-one** [comment, marked as answer]
DId you enable transaction logging on the server?

**Reply by well0886** [reply]
No, I didn't. I'll try to enable transaction logging on the server. Thank you for your help.

**Reply by well0886** [reply]
Yes, it works now. Thank you again for your help.
